### PR TITLE
Add hidden password field on share to fool the browser password-popup

### LINKED ISSUE #28259: Browser asking to save password when creating public link and new user
### Steps to reproduce
1. Create a public link with password
2. Change the folder

Screen capture:
password_issue

The same happends if you create a new user.

### Expected behaviour
No request to save password, because it doesn´t make sense.

### Actual behaviour
Browser is asking to save password.

### Server configuration
**Operating system**:

**Web server:**

**Database:**

**PHP version:**

**ownCloud version:** (see ownCloud admin page)
10.0.1
**Updated from an older ownCloud or fresh install:**

**Where did you install ownCloud from:**

**Signing status (ownCloud 9.0 and above):**

```

### PR DESCRIPTION
**Pull request by lordelix**
## Description
Added an extra hidden password field in the public share to fool the browsers password-save popup panel upon entering a password for the share.

Based on the ideas mentioned [here](https://gist.github.com/runspired/b9fdf1fa74fc9fb4554418dea35718fe)

## Related Issue

- Fixes https://github.com/owncloud/core/issues/28259

## How Has This Been Tested?

Visual test on Mac Chrome, Firefox and Safari as well as Win IE11

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

**Comment by felixheidecke** [comment]
This might need some testing on other machines and browsers.

File: core/js/sharedialoglinkshareview.js [hunk 1]
```diff
@@ -41,7 +41,8 @@
 			'{{/if}}' +
 			'<div id="linkPass-{{cid}}" class="public-link-modal--item linkPass">' +
 				'<label class="public-link-modal--label" for="linkPassText-{{cid}}">{{passwordLabel}}</label>' +
-				'<input class="public-link-modal--input linkPassText" id="linkPassText-{{cid}}" type="password" placeholder="{{passwordPlaceholder}}" />' +
+				'<input style="position:absolute;left:-100px;top:-100px;" type="password" name="fakepasswordrememberedworkaround">' +
```
Comment (PVince81) [inline core/js/sharedialoglinkshareview.js]: can we define a global CSS class for "offscreen" elements ? I thought we already had one

**Comment by PVince81** [comment]
@felixheidecke mind also testing Safari ?

File: core/js/sharedialoglinkshareview.js (hunk 1 above)
Comment (felixheidecke) [reply core/js/sharedialoglinkshareview.js]: I'll have a look, if I can't find any, I'll have one done.

**Comment by felixheidecke** [comment]
> mind also testing Safari ?

Works as well, at least on my system ;-) Would be nice to have more people test it.

**Comment by felixheidecke** [comment]
@haukman did you have time to check it out?

**Comment by haukman** [comment]
When I try to apply the patch I get the following error:

```
root@61eb98ec7331: /var/www/owncloud/core/js # patch < /var/www/owncloud/patch  --dry-run
(Patch is indented 2 spaces.)
checking file sharedialoglinkshareview.js
patch: **** malformed patch at line 23:                                   '</div>' +
```
Not sure if the issue is on my side ;)

**Review by PVince81 — CHANGES_REQUESTED** [review]

**Comment by felixheidecke** [comment]
@haukman … hmmm … "funny"

```
Applying: Add hidden password field on share to fool the browser password-popup
.git/rebase-apply/patch:14: trailing whitespace.
				'<input style="position:absolute;left:-100px;top:-100px;" type="password" name="fakepasswordrememberedworkaround">' +
error: patch failed: core/js/sharedialoglinkshareview.js:41
error: core/js/sharedialoglinkshareview.js: patch does not apply
Patch failed at 0001 Add hidden password field on share to fool the browser password-popup```

**Comment by felixheidecke** [comment]
@PVince81 I added the class. However, this "fix" is very hacky and wobbly and may not work in upcoming browser releases.

**Comment by felixheidecke** [comment]
@haukman have you found other means of testing this? Can't help you with the patch :-(

**Comment by patrickjahns** [comment]
@ownclouders rebase

**Comment by ownclouders** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

**Comment by ownclouders** [comment]
Automated rebase with [GitMate.io](https://gitmate.io) was successful! :tada:

**Comment by haukman** [comment]
@PVince81: Can you help with the patch?

**Comment by PVince81** [comment]
@felixheidecke please recreate a patch for @haukman based on stable10 or v10.0.9.
a patch of a PR to master likely doesn't apply cleanly on stable10 or v10.0.9

**Comment by felixheidecke** [comment]
The patch is the same for stable10: [32195-stable10.txt](https://github.com/owncloud/core/files/2282939/32195-stable10.patch.txt)

**Comment by DeepDiver1975** [comment]
> The patch is the same for stable10: 32195-stable10.txt

@haukman can you test and verify this is fixed now? THX

**Comment by PVince81** [comment]
@felixheidecke please create a patch on top of v10.0.9 not stable10.

**Comment by haukman** [comment]
same issue as before: 

```
root@ae1a8e355dc9: /var/www/owncloud # patch -p 1 <patch 
patching file core/js/sharedialoglinkshareview.js
patch: **** malformed patch at line 13:                                   '</div>' +
```
Did you test that patch? :P

**Comment by felixheidecke** [comment]
@haukman have you simply tried `git apply 32195-v10.0.9.patch.txt` ??
I see no need for the -p flag here

**Comment by PVince81** [comment]
@felixheidecke he's using the tarball, please provide a patch and a command that works for the 10.0.9 tarball

**Comment by felixheidecke** [comment]
Sorry @haukman the patch works fine on my system. I just patched my 10.0.9 tarball:

bildschirmfoto von 2018-08-30 10-43-40

Are you using Windows or some sh*t?

**Comment by felixheidecke** [comment]
What worked the last time doesn't work anymore :-(
There is another workaround that kinda goes like this':

`<input name="password" type="text" style="font-family:'dotsfont'>` 

Mean that the password field is reverted to a simple text input using a font-face that only displays dots. I don't know if this threatens security or not. If the font doesn't load, the password is displayed.

Here is the font in question: https://github.com/kylewelsby/dotsfont
Is this so important that we want to go this route?

**Comment by PVince81** [comment]
@settermjd any thoughts about this approach ?

This is likely not an issue because the password is being typed and any JS code can read it anyway, regardless of field type.

This also means that selecting the dots and copy-pasting will work as well.

@felixheidecke do all modern browsers support loading custom fonts now ? Has this approach been recommended anywhere ?

**Comment by PVince81** [comment]
does the font also work for Chinese 汉字 and Cyrillic for example ?

**Comment by felixheidecke** [comment]
There might be another solution to the problem. We cloud forego the classic form submission so that the browser does not recognize the password saving process. I'll give it a go and come back with the results.

**Comment by felixheidecke** [comment]
As I was looking into it, there is no classic `.submit()` anywhere and there is even no `<form>` wrapping around it. So the solution I suggested is not valid.

At this point I would just leave it alone and live with it. *…sigh*

**Comment by felixheidecke** [comment]
@haukman any updates on how to proceed? I wouldn't want to sink any more time into this sinking ship :-P

**Comment by felixheidecke** [comment]
After further research, I found no solution that works reliably across all browsers. I suggest keeping an eye on the matter and I will open a new PR / issue if browsers do change in that matter.

**Pull request closed without merging**